cmd/sriovdp/manager.go

```diff
@@ -117,19 +119,76 @@ func (rm *resourceManager) initServers() error {
 
 		devices := dp.GetDevices(rc)
 		filteredDevices, err := dp.GetFilteredDevices(devices, rc)
-		if err != nil {
-			glog.Errorf("initServers(): error getting filtered devices for config %+v: %q", rc, err)
-		}
+
 		filteredDevices = rm.excludeAllocatedDevices(filteredDevices, deviceAllocated)
+
 		if len(filteredDevices) < 1 {
 			glog.Infof("no devices in device pool, skipping creating resource server for %s", rc.ResourceName)
 			continue
 		}
+
 		rPool, err := rm.rFactory.GetResourcePool(rc, filteredDevices)
 		if err != nil {
 			glog.Errorf("initServers(): error creating ResourcePool with config %+v: %q", rc, err)
 			return err
 		}
+
+		if rm.useCdi {
+
+			cdiDevices := make([]specs.Device, 0)
+			cdiSpec := specs.Spec{
+				Version: "0.5.0",
+				Kind:    rm.resourcePrefix + "/net",
+				Devices: cdiDevices,
+			}
+
+			for _, dev := range filteredDevices {
+				containerEdit := specs.ContainerEdits{
+					DeviceNodes: make([]*specs.DeviceNode, 0),
+					Env:         make([]string, 0),
+				}
+
+				envs, err := rPool.GetEnvs(rm.resourcePrefix, []string{dev.GetDeviceID()})
+				if err != nil {
+					glog.Errorf("Unable to create environment variables: %s", err)
+					return fmt.Errorf("Unable to create environment variables")
+				}
+				for name, value := range envs {
+					containerEdit.Env = append(containerEdit.Env, name+"="+value)
+				}
+
+				for i, spec := range dev.GetDeviceSpecs() {
+					glog.Infof("Checking device spec (%d): %s", i, spec.String())
+					deviceNode := specs.DeviceNode{
```

Comment:
can you rename either specs to cdiSpecs in imports or spec in the range statement above. its confusing to read as the two refer to two different things.

Reply:
done